PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/sqlserver/creation/SqlServerCreationAdvancedPanel.java

```diff
@@ -0,0 +1,155 @@
+import com.microsoft.azure.toolkit.intellij.mysql.AdminUsernameTextField;
+import com.microsoft.azure.toolkit.intellij.mysql.ConnectionSecurityPanel;
+import com.microsoft.azure.toolkit.intellij.mysql.creation.PasswordUtils;
+import com.microsoft.azure.toolkit.intellij.sqlserver.common.ServerNameTextField;
+import com.microsoft.azure.toolkit.intellij.sqlserver.common.SqlServerNameTextField;
+import com.microsoft.azure.toolkit.intellij.sqlserver.common.SqlServerRegionComboBox;
+import com.microsoft.azure.toolkit.lib.common.form.AzureFormInput;
+import com.microsoft.azure.toolkit.lib.sqlserver.SqlServerConfig;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.swing.*;
+import javax.swing.event.DocumentListener;
+import java.awt.event.ItemEvent;
+import java.util.Arrays;
+import java.util.List;
+
+public class SqlServerCreationAdvancedPanel extends JPanel implements AzureFormPanel<SqlServerConfig> {
+
+    private JPanel rootPanel;
+    private ConnectionSecurityPanel security;
+    @Getter
+    private SubscriptionComboBox subscriptionComboBox;
+    @Getter
+    private ResourceGroupComboBox resourceGroupComboBox;
+    @Getter
+    private SqlServerNameTextField serverNameTextField;
+    @Getter
+    private SqlServerRegionComboBox regionComboBox;
+    @Getter
+    private AdminUsernameTextField adminUsernameTextField;
+    @Getter
+    private JPasswordField passwordField;
+    @Getter
+    private JPasswordField confirmPasswordField;
+
+    private AzurePasswordFieldInput passwordFieldInput;
+    private AzurePasswordFieldInput confirmPasswordFieldInput;
+
+    private SqlServerConfig config;
+
+    SqlServerCreationAdvancedPanel(SqlServerConfig config) {
+        super();
+        this.config = config;
+        $$$setupUI$$$(); // tell IntelliJ to call createUIComponents() here.
+        init();
+        initListeners();
+        setData(config);
+    }
+
+    private void init() {
+        passwordFieldInput = PasswordUtils.generatePasswordFieldInput(this.passwordField, this.adminUsernameTextField);
+        confirmPasswordFieldInput = PasswordUtils.generateConfirmPasswordFieldInput(this.confirmPasswordField, this.passwordField);
+    }
+
+    private void initListeners() {
+        this.subscriptionComboBox.addItemListener(this::onSubscriptionChanged);
+        this.adminUsernameTextField.getDocument().addDocumentListener(generateAdminUsernameListener());
+    }
+
+    private void onSubscriptionChanged(final ItemEvent e) {
+        if (e.getStateChange() == ItemEvent.SELECTED && e.getItem() instanceof Subscription) {
+            final Subscription subscription = (Subscription) e.getItem();
+            this.resourceGroupComboBox.setSubscription(subscription);
+            this.serverNameTextField.setSubscriptionId(subscription.subscriptionId());
+            this.regionComboBox.setSubscription(subscription);
+        }
+    }
+
+    private DocumentListener generateAdminUsernameListener() {
```

Comment:
no need to separate a method generateAdminUsernameListener

Reply:
it's only a private method. I think it should be ok.
If not, the init() method will have a lot of code lines, isn't it?

Reply:
merge all listeners into only one method? I think it not a good idea.

![image](https://user-images.githubusercontent.com/19339116/118383309-41e24c80-b62f-11eb-84f3-b22a288b7892.png)

Separate it to two methods is better than seting them in one method. isn't it ?
and It maybe add more in the future.

Reply:
renamed.